blockstore/src/main/java/brooklyn/location/blockstore/NewVolumeCustomizer.java

```diff
@@ -3,25 +3,108 @@
 import brooklyn.location.blockstore.api.MountedBlockDevice;
 import brooklyn.location.blockstore.api.VolumeManager;
 import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.reflect.TypeToken;
+import org.apache.brooklyn.config.ConfigKey;
+import org.apache.brooklyn.core.config.ConfigKeys;
 import org.apache.brooklyn.location.jclouds.BasicJcloudsLocationCustomizer;
+import org.apache.brooklyn.location.jclouds.JcloudsLocation;
 import org.apache.brooklyn.location.jclouds.JcloudsMachineLocation;
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.jclouds.compute.ComputeService;
 import org.jclouds.compute.domain.NodeMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.util.List;
 import java.util.Map;
 
 public abstract class NewVolumeCustomizer extends BasicJcloudsLocationCustomizer {
-    protected Map<BlockDeviceOptions, FilesystemOptions> volumes;
 
-    protected MountedBlockDevice mountedBlockDevice;
+    private static final Logger LOG = LoggerFactory.getLogger(NewVolumeCustomizer.class);
 
-    public abstract Map<BlockDeviceOptions, FilesystemOptions> getVolumes();
+    public static final ConfigKey<List<Map<?, ?>>> VOLUMES = ConfigKeys.newConfigKey(
+            new TypeToken<List<Map<?, ?>>>() {},
+            "volumes", "List of volumes to be attached");
 
-    public abstract MountedBlockDevice getMountedBlockDevice();
+    /**
+     * Used only for checking results from customization
+     */
+    protected transient MountedBlockDevice mountedBlockDevice;
+
+    protected NewVolumeCustomizer() {
+    }
+
+    public NewVolumeCustomizer(Map<BlockDeviceOptions, FilesystemOptions> volume) {
+        setVolumes(ImmutableList.<Map<?, ?>>of(volume));
```

Comment:
`setVolumes` expected YAML-style untyped map, while you are passing in `Map<BlockDeviceOptions, FilesystemOptions>`. Doesn't look right.